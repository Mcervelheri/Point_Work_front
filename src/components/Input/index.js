import React, {
    lazy, Suspense, forwardRef, memo, useCallback,
} from 'react';

import {
    Input as AntdInput, Form, Switch, Checkbox, InputNumber,
} from 'antd';
import omit from 'lodash/omit';

import { wrapFormField } from '../helpers/form-helper';

const DatePicker = lazy(() => import('../DatePicker'));
const Select = lazy(() => import('../Select'));
const DateRangePicker = lazy(() => import('../DateRangePicker'));
const InputMinMax = lazy(() => import('../InputMinMax'));

const { TextArea, Search } = AntdInput;
const { Item } = Form;

const Input = memo(forwardRef((props, ref) => {
    const {
        input, meta,
        label, formItemProps,
        required, hasFeedback,
        hasFormItem, charCounter, maxLength,
        errorMessage, help, alwaysShowErrorMessage,
    } = props;

    const { onChange, type, value } = input;

    const {
        valid, touched, error, submitError,
    } = meta;

    const displayError = Boolean(alwaysShowErrorMessage || touched);

    const handleDateChange = useCallback(newValue => {
        if (onChange) onChange(newValue);
    }, [onChange]);

    const getValidateStatus = () => {
        if (!displayError) return null;

        return valid ? 'success' : 'error';
    };

    const renderTextInput = inputProps => {

        return (
            <AntdInput
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderTextArea = inputProps => {
        return (
            <TextArea
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderSearch = inputProps => {
        return (
            <Search
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderDatePicker = inputProps => {
        const {
            dateFormat,
            ...others
        } = inputProps;
        return (
            <DatePicker
                {...others}
                {...input}
                ref={ref}
                onChange={handleDateChange}
                format={dateFormat}
            />
        );
    };

    const renderSelect = inputProps => {
        return (
            <Select
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderDateRangePicker = inputProps => {
        const {
            dateFormat,
            ...others
        } = inputProps;
        return (
            <DateRangePicker
                {...others}
                {...input}
                ref={ref}
                onChange={handleDateChange}
                format={dateFormat}
            />
        );
    };

    const renderSwitch = inputProps => {
        return (
            <Switch
                {...inputProps}
                {...input}
                ref={ref}
                value={null}
                checked={value || false}
            />
        );
    };

    const renderCheckbox = inputProps => {
        return (
            <Checkbox
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderCheckboxGroup = inputProps => {
        return (
            <Checkbox.Group
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderInputNumber = inputProps => {
        return (
            <InputNumber
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderInputMinMax = inputProps => {
        return (
            <InputMinMax
                {...inputProps}
                {...input}
                ref={ref}
            />
        );
    };

    const renderInputComponent = () => {

        const inputProps = omit(props, [
            'label', 'formItemProps', 'charCounter',
            'hasFormItem', 'meta', 'input', 'parser',
            'hasFeedback', 'alwaysShowErrorMessage',
        ]);

        switch (type) {
            case 'date':
                return renderDatePicker(inputProps);
            case 'textarea':
                return renderTextArea(inputProps);
            case 'select':
                return renderSelect(inputProps);
            case 'daterange':
                return renderDateRangePicker(inputProps);
            case 'switch':
                return renderSwitch(inputProps);
            case 'checkbox':
                return renderCheckbox(inputProps);
            case 'checkbox-group':
                return renderCheckboxGroup(inputProps);
            case 'search':
                return renderSearch(inputProps);
            case 'number':
                return renderInputNumber(inputProps);
            case 'minmax':
                return renderInputMinMax(inputProps);
            default:
                return renderTextInput(inputProps);
        }
    };

    if (!hasFormItem) return renderInputComponent();

    const renderHelp = () => {

        if (displayError && !valid) {
            return errorMessage || error || submitError;
        }

        if (charCounter) {
            const charCounterMessage = charCounter
                ? `${(value || '').length} / ${maxLength}`
                : null;
            return charCounterMessage;
        }

        return help;
    };

    return (
        <Suspense fallback="Carregando...">
            <Item
                label={label}
                help={renderHelp()}
                hasFeedback={!meta.active && meta.touched && hasFeedback}
                validateStatus={getValidateStatus()}
                required={required}
                {...formItemProps}
            >
                {renderInputComponent()}
            </Item>
        </Suspense>
    );

}));

Input.defaultProps = {
    meta: {},
    hasFormItem: true,
};

Input.Field = wrapFormField(Input);

export default Input;
