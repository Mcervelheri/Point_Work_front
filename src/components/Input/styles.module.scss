@import "../theme/colors.scss";

:global(.ant-form-item-has-error) {
    .reactMde {
        border-color: $color-red-6;
    }
}

.reactMdeTextArea {
    line-height: 1.41em;
}
