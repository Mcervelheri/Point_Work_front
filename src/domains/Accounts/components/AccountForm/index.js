import React, {
    useCallback, useState, useMemo, useEffect,
} from 'react';

import {
    SaveOutlined, EditOutlined,
} from '@ant-design/icons';
import {
    Button, Row, Col,
} from 'antd';

import Form from '../../../../components/Form';
import LoadingSpin from '../../../../components/LoadingSpin';
import { extractRequestError } from '../../../../helpers/error-helper';
import { wrapForm } from '../../../../helpers/form-helper';
import useAsync from '../../../../hooks/use-async';
import useZipCode from '../../../../hooks/use-zip-code';

import DocumentsUpload from './components/DocumentsUpload';
import styles from './styles.module.scss';

const formAddressPhysical = {
    bairro: 'mainAddressNeighborhood',
    cidade: 'mainAddressCity',
    endereco: 'mainAddressStreet',
    uf: 'mainAddressState',
    country: 'mainAddressCountry',
};

const formAddressLegal = {
    bairro: 'pj__mainAddressNeighborhood',
    cidade: 'pj__mainAddressCity',
    endereco: 'pj__mainAddressStreet',
    uf: 'pj__mainAddressState',
    country: 'pj__mainAddressCountry',
};

const FormComp = wrapForm(props => {

    const {
        handleSubmit, onSave, onEdit,
        isInsertionMode, children,
        data, form, company, hasValidationErrors,
    } = props;

    const [initialized, setInitialized] = useState(false);

    const { loading: loadingZipCode } = useZipCode(
        form,
        company ? 'pj__mainAddressZipCode' : 'mainAddressZipCode',
        company ? formAddressLegal : formAddressPhysical,
    );

    const renderChildren = () => {
        if (typeof children === 'function') {
            return children(isInsertionMode, loadingZipCode);
        }
        return children;
    };

    useEffect(() => {
        if (data && !initialized) {
            setInitialized(true);
            form.initialize(data);
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [data]);

    const { call: onSubmit, loading } = useAsync(async values => {

        try {
            await onSave(values);
        } catch (exception) {
            extractRequestError(exception);
        }
    }, [onSave]);

    const renderButtonSubmit = () => {
        if (!isInsertionMode) return null;

        return (
            <div className={styles.buttonSubmit}>
                <Button
                    type="primary"
                    htmlType="submit"
                    disabled={hasValidationErrors}
                    icon={<SaveOutlined />}
                    loading={loading}
                >
                    Salvar
                </Button>
            </div>
        );
    };

    return (
        <>
            <Form
                className={styles.form}
                onSubmit={handleSubmit(onSubmit)}
                key={isInsertionMode}
            >
                {renderChildren()}

                {renderButtonSubmit()}
            </Form>

            {!isInsertionMode && (
                <div className={styles.buttonEdit}>
                    <Button
                        type="primary"
                        onClick={onEdit}
                        icon={<EditOutlined />}
                    >
                        Editar
                    </Button>
                </div>
            )}
        </>
    );
});

const AccountForm = props => {

    const {
        onSave, onDelete,
        children, data, hasLog, company, idAccount, parentLoadingArr = [], createMode,
        idClient, documents,
    } = props;

    const [isInsertionMode, setIsInsertionMode] = useState(createMode);
    const [loadingDocumentsUpload, setLoadingDocumentsUpload] = useState(false);

    const handleEdit = useCallback(() => {
        setIsInsertionMode(true);
    }, []);

    const loadingArr = useMemo(() => {
        return [...parentLoadingArr, loadingDocumentsUpload];
    }, [parentLoadingArr, loadingDocumentsUpload]);

    return (
        <>
            <LoadingSpin loadingArr={loadingArr} />
            <div className={styles.container}>
                <Row gutter={12}>
                    <Col
                        sm={24}
                        lg={isInsertionMode ? 12 : 24}
                        className={styles.formCol}
                    >
                        <FormComp
                            hasLog={hasLog}
                            onSave={onSave}
                            onDelete={onDelete}
                            company={company}
                            onEdit={handleEdit}
                            isInsertionMode={isInsertionMode}
                            data={data}
                        >
                            {children}
                        </FormComp>
                    </Col>
                    {isInsertionMode && (
                        <Col sm={24} lg={12}>
                            <DocumentsUpload
                                idAccount={idAccount}
                                idClient={idClient}
                                company={company}
                                setLoading={setLoadingDocumentsUpload}
                                documents={documents}
                                createMode={createMode}
                            />
                        </Col>
                    )}
                </Row>
            </div>
        </>
    );
};

export default AccountForm;
