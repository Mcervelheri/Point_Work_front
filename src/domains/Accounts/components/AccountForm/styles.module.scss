@import "../../../../theme/colors.scss";

.container {
    padding: 48px 99px 0px 99px;
}

.form {
    max-width: 395px;
}

.formCol {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.buttonSubmit {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    margin-top: 34px;
}

.buttonEdit {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    margin-top: 34px;
    width: 360px;
}

.buttonEditWithoutLog {
    margin-left: 300px;
    margin-top: 34px;
}

.iconCheck {
    color: $color-green-6 !important;
}
