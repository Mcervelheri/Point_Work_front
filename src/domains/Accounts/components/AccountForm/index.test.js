import React from 'react';

import { create } from 'react-test-renderer';

import Input from '../components/Input';

import AccountForm from './index';

const FORM_MOCK = {
    legalName: 'Empresa Fake',
};

const inputProps = {
    onChange: jest.fn,
    value: 'Razão Social',
};

test('renderiza componente <AccountForm /> no modo de visualização corretamente', () => {
    const tree = create(
        <AccountForm
            data={FORM_MOCK}
            onSave={jest.fn}
        >
            <Input
                name="legalName"
                label="Razão Social"
                input={inputProps}
            />
        </AccountForm>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renderiza componente <AccountForm /> no modo de inserção corretamente', () => {
    const tree = create(
        <AccountForm
            onSave={jest.fn}
        >
            <Input
                name="legalName"
                label="Razão Social"
                input={inputProps}
            />
        </AccountForm>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renderiza componente <AccountForm /> com opção para deleção corretamente', () => {
    const tree = create(
        <AccountForm
            onDelete={jest.fn}
        >
            <Input
                name="legalName"
                label="Razão Social"
                input={inputProps}
            />
        </AccountForm>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renderiza componente <AccountForm /> com conteúdo do children como função corretamente', () => {
    const tree = create(
        <AccountForm
            onDelete={jest.fn}
        >
            {() => (
                <Input
                    name="legalName"
                    label="Razão Social"
                    input={inputProps}
                />
            )}
        </AccountForm>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
