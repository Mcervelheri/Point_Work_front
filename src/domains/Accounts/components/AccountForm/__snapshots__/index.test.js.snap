// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza componente <AccountForm /> com conteúdo do children como função corretamente 1`] = `
<div
  className="container"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -6,
        "marginRight": -6,
      }
    }
  >
    <div
      className="ant-col formCol ant-col-sm-24 ant-col-lg-24"
      style={
        Object {
          "paddingLeft": 6,
          "paddingRight": 6,
        }
      }
    >
      <div
        className="ant-form ant-form-vertical form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="ant-form-item"
          >
            <div
              className="ant-row ant-form-item-row"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  title="Razão Social"
                >
                  Razão Social
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      name="legalName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      type="text"
                      value="Razão Social"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="buttonEdit"
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="edit"
            className="anticon anticon-edit"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="edit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </span>
          <span>
            Editar
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renderiza componente <AccountForm /> com opção para deleção corretamente 1`] = `
<div
  className="container"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -6,
        "marginRight": -6,
      }
    }
  >
    <div
      className="ant-col formCol ant-col-sm-24 ant-col-lg-24"
      style={
        Object {
          "paddingLeft": 6,
          "paddingRight": 6,
        }
      }
    >
      <div
        className="ant-form ant-form-vertical form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="ant-form-item"
          >
            <div
              className="ant-row ant-form-item-row"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  title="Razão Social"
                >
                  Razão Social
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      name="legalName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      type="text"
                      value="Razão Social"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="buttonEdit"
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="edit"
            className="anticon anticon-edit"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="edit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </span>
          <span>
            Editar
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renderiza componente <AccountForm /> no modo de inserção corretamente 1`] = `
<div
  className="container"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -6,
        "marginRight": -6,
      }
    }
  >
    <div
      className="ant-col formCol ant-col-sm-24 ant-col-lg-24"
      style={
        Object {
          "paddingLeft": 6,
          "paddingRight": 6,
        }
      }
    >
      <div
        className="ant-form ant-form-vertical form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="ant-form-item"
          >
            <div
              className="ant-row ant-form-item-row"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  title="Razão Social"
                >
                  Razão Social
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      name="legalName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      type="text"
                      value="Razão Social"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="buttonEdit"
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="edit"
            className="anticon anticon-edit"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="edit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </span>
          <span>
            Editar
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renderiza componente <AccountForm /> no modo de visualização corretamente 1`] = `
<div
  className="container"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -6,
        "marginRight": -6,
      }
    }
  >
    <div
      className="ant-col formCol ant-col-sm-24 ant-col-lg-24"
      style={
        Object {
          "paddingLeft": 6,
          "paddingRight": 6,
        }
      }
    >
      <div
        className="ant-form ant-form-vertical form"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="ant-form-item"
          >
            <div
              className="ant-row ant-form-item-row"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  title="Razão Social"
                >
                  Razão Social
                </label>
              </div>
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      name="legalName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      type="text"
                      value="Razão Social"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="buttonEdit"
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="edit"
            className="anticon anticon-edit"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="edit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </span>
          <span>
            Editar
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
